processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -251,6 +245,112 @@ public static boolean convertSegment(File toConvert, File converted, IndexSpec i
     public MMappedIndex mapDir(File inDir) throws IOException;
   }
 
+  public static void validateRowValues(
+      Rowboat rb1,
+      IndexableAdapter adapter1,
+      Rowboat rb2,
+      IndexableAdapter adapter2
+  )
+  {
```

Comment:
any reason we are not validating
rb1.getTimestamp() == rb2.getTimestamp()
in this code or is that not required?